test/controller.bar.tests.js

```diff
@@ -487,4 +489,154 @@ describe('Bar controller tests', function() {
 		expect(bar._model.borderColor).toBe('rgb(0, 255, 0)');
 		expect(bar._model.borderWidth).toBe(1.5);
 	});
+
+	describe('Bar width', function() {
+		beforeEach(function() {
+			// 2 datasets
+			this.data = {
+				labels: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
+				datasets: [{
+					data: [10, 20, 30, 40, 50, 60, 70],
+				}, {
+					data: [10, 20, 30, 40, 50, 60, 70],
+				}]
+			};
+		});
+
+		afterEach(function() {
```

Comment:
Oh... so..  something like this?
(I have not tested this code. It is an image.)

``` javascript

beforeEach(function() {
	this.data = {
		labels: ['January', 'February', 'March', 'April', 'May', 'June', 'July'],
		datasets: [{
			data: [10, 20, 30, 40, 50, 60, 70],
		}, {
			data: [10, 20, 30, 40, 50, 60, 70],
		}]
	};

	this function getTotalBarHeight(function() {
		var chart = window.acquireChart(this.config);
		var meta = chart.getDatasetMeta(0);
		var yScale = chart.scales[meta.yAxisID];

		var categoryPercentage = yScale.options.categoryPercentage;
		var barPercentage = yScale.options.barPercentage;
		var stacked = yScale.options.stacked;

		var totalBarHeight = 0;
		for (var i = 0; i < chart.data.datasets.length; i++) {
			var bars = chart.getDatasetMeta(i).data;
			for (var j = yScale.minIndex; j <= yScale.maxIndex; j++) {
				totalBarHeight += bars[j]._model.height;
			}
			if (stacked) {
				break;
			}
		}

		return {
			actual:totalBarHeight,
			expected:yScale.height * categoryPercentage * barPercentage,
		}

	};

});

it('should correctly set bar height when min and max option is set.', function() {
	this.config = {
		type: 'horizontalBar',
		data: this.data,
		options: {
			scales: {
				yAxes: [{
					ticks: {
						min: 'March',
						max: 'May',
					},
				}]
			}
		}
	};
	var testValue = this.getTotalBarHeight();
	expect(testValue.actual).toBeCloseToPixel(testValue.expected);
});

it('should correctly set bar height when scale are stacked with min and max options.', function() {
	this.config = {
		type: 'horizontalBar',
		data: this.data,
		options: {
			scales: {
				yAxes: [{
					ticks: {
						min: 'March',
						max: 'May',
					},
					stacked: true,
				}]
			}
		}
	};
	var testValue = this.getTotalBarHeight();
	expect(testValue.actual).toBeCloseToPixel(testValue.expected);
});


```